graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,387 @@
+            raise ValueError(msg)
+        if len(seedsA) != len(seedsB):
+            msg = "Must have the same number of seeds for each adjacency matrix"
+            raise ValueError(msg)
+        if len(seedsA) == 0:
+            msg = "Must include at least one seed to produce nomination list"
+            raise ValueError(msg)
+
+        # get vertex reordering for Ax
+        # in the form (seedsA, voi, rest in order)
+        nsx1 = np.setdiff1d(np.arange(A.shape[0]), np.append(seedsA, voi))
+        a_reord = np.append(np.append(seedsA, voi), nsx1)
+
+        # get reordering for B in the form (seedsB, rest in numerical order)
+        nsx2 = np.setdiff1d(np.arange(B.shape[0]), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        # Reorder the two graphs with our new vertices order
+        AA = A[a_reord][:, a_reord]
+        BB = B[b_reord][:, b_reord]
+
+        # Record where the new seeds and voi locations are
+        # in our re-ordered graphs
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+
+        # Determine what seeds are within a specified subgraph
+        # given by `self.order_voi_subgraph`. If there are no
+        # seeds in this subgraph, print a message and return None
+        subgraph_AA = _get_induced_subgraph_list(
+            AA, self.order_voi_subgraph, voi_reord, mindist=1
+        )
+
+        close_seeds = np.intersect1d(subgraph_AA, seeds_reord)
+
+        if len(close_seeds) <= 0:
+            warnings.warn(
+                "Voi {} was not a member of the induced subgraph A[{}], \
+                Try increasing order_voi_subgraph".format(
+                    voi, seedsA
+                )
+            )
+            self.n_seeds_ = None
+            self.nomination_list_ = None
+            return self
+
+        # Generate the two induced subgraphs that will be used by the matching
+        # algorithm using the seeds that we identified in the previous step.
+        verts_A = _get_induced_subgraph_list(
+            AA, self.order_seeds_subgraph, close_seeds, mindist=0
+        )
+
+        verts_B = _get_induced_subgraph_list(
+            BB, self.order_seeds_subgraph, close_seeds, mindist=0
+        )
+
+        # Determine the final reordering for the graphs that include only
+        # the vertices found by the induced subgraphs in the previous step
+        # For graph A, its of the form (close_seeds, voi, rest in verts_A
+        # in num order). For graph B its of the form (close_seeds, rest in
+        # verts_B in num order)
+        paramed_verts = np.append(close_seeds, voi_reord)
+        ind1 = np.append(
+            np.append(close_seeds, voi_reord), np.setdiff1d(verts_A, paramed_verts)
+        )
+        ind2 = np.concatenate((close_seeds, np.setdiff1d(verts_B, close_seeds)))
+
+        # Generate adjacency matrices for the ordering found in the prev step
+        AA_fin = AA[ind1][:, ind1]
```

Comment:
this is SG_1 from docs right? maybe call it that, but give it a more informative name in general